web/src/index.ts

```diff
@@ -91,47 +94,40 @@ export const main = (
   function initComponents(): void {
     window.removeEventListener('appload', initComponents)
 
-    // `executeHandler` for `CodeChunk` and `CodeExpression` nodes
-    window.document
-      .querySelectorAll<
-        HTMLStencilaCodeChunkElement | HTMLStencilaCodeExpressionElement
-      >('stencila-code-chunk,stencila-code-expression')
-      .forEach((elem) => {
-        elem.executeHandler = async <C extends CodeChunk | CodeExpression>(
-          node: C
-        ): Promise<C> => {
-          const [client, document] = await startup()
-          await documents.execute(client, document.id, elem.id)
-          // The WebComponent for a `CodeExpression` has a `isOutputEmpty` property
-          // which is set based on the return value from this function and does not
-          // change later when we actually update the output. So, here's a hack to
-          // make that always true.
-          return Promise.resolve({ ...node, output: '' })
-        }
-      })
-
-    // Temporary functions for testing in the console
-    // @ts-ignore
-    window.stencilaExecute = async (
-      nodeId: null | string,
-      ordering: 'Single' | 'Appearance' | 'Topological'
-    ) => {
+    // Code execution
+    const executeHandler = async ({ detail }: CodeExecuteEvent) => {
       const [client, document] = await startup()
-      await documents.execute(client, document.id, nodeId, ordering)
+      await documents.execute(
+        client,
+        document.id,
+        detail.nodeId,
+        detail.ordering
+      )
     }
-    // @ts-ignore
-    window.stencilaCancel = async (
-      nodeId: null | string,
-      scope: 'Single' | 'All'
-    ) => {
+
+    window.addEventListener('stencila-code-execute', (e) => {
```

Comment:
@alex-ketch Is there any particular reason these `window.addEventListener` calls are here and not in `documents.listen`? It feels like having them all in one place - whichever is the best place - would make the code more maintainable. If there is a reason listeners for `stencila-code-execute`, `stencila-code-execute-cancel`, `stencila-kernel-restart` need to be here and not in the `documents.listen` could you document that.

Reply:
nope, was focussing on refactoring the `index.ts` file and didn't see that the event listeners were grouped in the `documents.ts` file.
Refactoring now